File: sem-1/wdp-zadania-main/lab9/zad1.py
def bubbleSort(arr: list[int]):
    j: int = len(arr) - 1

    while j >= 1:
        i = 0
        while i < j:
            if arr[i] > arr[i + 1]:
                arr[i], arr[i+1] = arr[i+1], arr[i]
            i += 1
        j -= 1
    return arr


print(bubbleSort([8, 4, 6, 7]))

File: sem-2/jp1/lab5/zad1.scala
package lab5

import scala.annotation.tailrec

def cleanUp[T](arr: List[T]): List[T] = {
  @tailrec
  def helper(arr: List[T], acc: List[T] = Nil): List[T] = arr match {
    case Nil => acc.reverse
    case head :: Nil => (head :: acc).reverse
    case first :: second :: tail if first == second => helper(second :: tail, acc)
    case first :: second :: tail => helper(second :: tail, first :: acc)
  }

  helper(arr)
}

@main
def lab5zad1(): Unit = {
  println(cleanUp(List(1, 1, 2, 4, 4, 4, 1, 3)))
}
